src/test/unit/engine/NetworkTest.cpp

```diff
@@ -368,7 +344,6 @@ TEST(NetworkTest, Phases) {
   ASSERT_TRUE(phaseSet.size() == 1);
   ASSERT_TRUE(phaseSet.find(1) != phaseSet.end());
 
-  EXPECT_THROW(net.initialize(), std::exception);
```

Comment:
should be removed?

Reply:
> should be removed?

yes, net.initialize does not need to have dimensions set before it can execute anymore.  It would throw an exception only if it could not determine the dimensions on all connected inputs/outputs.   I think I test that condition someplace...

This is not a change in API since it was ONLY the TestNode that would throw if it did not have dimensions explicitly defined.  Now, this is done in the base class so its covered too.